File: smart-poultry-dashboard/src/types/sensor.ts
import { SensorType } from "./enums";

export interface Sensor {
  id: string;
  pen_id: string;
  type: SensorType;
  device_id?: string | null;
  created_at: string;
}

export interface SensorCreate {
  pen_id: string;
  type: SensorType;
  device_id?: string | null;
}

File: app/schemas/sensor.py
from datetime import datetime
from pydantic import BaseModel
from app.models.enums import SensorTypeEnum

class SensorBase(BaseModel):
    type: SensorTypeEnum
    device_id: str | None = None  # MCU channel

class SensorCreate(SensorBase):
    pen_id: str

class SensorUpdate(BaseModel):
    device_id: str | None = None

class SensorOut(SensorBase):
    id: str
    pen_id: str
    created_at: datetime

    class Config:
        from_attributes = True
